superset/daos/query.py

```diff
@@ -35,25 +35,6 @@
 class QueryDAO(BaseDAO[Query]):
     base_filter = QueryFilter
 
-    @staticmethod
-    def update_saved_query_exec_info(query_id: int) -> None:
-        """
-        Propagates query execution info back to saved query if applicable
-
-        :param query_id: The query id
-        :return:
-        """
-        query = db.session.query(Query).get(query_id)
-        related_saved_queries = (
-            db.session.query(SavedQuery)
-            .filter(SavedQuery.database == query.database)
-            .filter(SavedQuery.sql == query.sql)
```

Comment:
Can't we just add `.filter(SavedQuery.id == query_id)` here? It should use the PK index and be super efficient. Also if we don't want to update if the query has been mutated, seems we should not only check `sql` and `database`, but schema/catalog too.

Reply:
I was super confused by these filters, too. Is the intent here to update all entries where the same database + query combo are featured? Seems very strange to want to do that..?

Reply:
Yeah it says "check all saved queries that have the same SQL, and update their last-executed time". Clearly if it's not the same SavedQuery.id, we shouldn't update the last executed time.

Reply:
Adding a filter on SavedQuery.id should solve the long execution time in environments with lots of SavedQueries

Reply:
But it seems to be very explicit about updating many rows, so I can only assume the intent here is to actually update many queries. Looking at #11391 it does indeed appear to be the intent. But this is clearly a perf problem, so I would do away with this feature, and just update the actual query that ran.

Reply:
Ping @dpgaspar if you recall why this was needed.

Reply:
btw the condition `.filter(SavedQuery.id == query_id)` is not correct.
To find the saved query that matches the current query, it is appropriate to find the `saved_query_id` of the `tab_state` that matches the `latest_query_id` and `client_id` of the current query.

Reply:
If this information is truly necessary, it would be better to update the saved query by making an additional PUT request after the execute action is performed on the **frontend** (especially when the current query editor's savedQueryId exists.), rather than handling it during the `sqllab/execute process`.

Reply:
Gotcha, I though the `query_id` in context here was a `SavedQuery.id` but it's actually a `Query.id` ...

Reply:
Even though we could do an additional PUT as @justinpark suggested, I still think showing the last time a saved query run is less important than showing the last time it was modified. That's what the other entities show and it's generally what users look for when they find a problem.

Reply:
I agree with @michael-s-molina  opinion. Unless there are significant reasons from @dpgaspar and others, I will prioritize merging this hotfix and will follow up with additional PRs if needed based on the remaining feedback.